src/sagemaker/workflow/pipeline.py

```diff
@@ -215,30 +215,29 @@ def upsert(
         Returns:
             response dict from service
         """
+        exists = True
         try:
+            self.describe()
+        except ClientError:
+            exists = False
```

Comment:
nit: is there a better way to check this?

Reply:
ListPipelines is another option but I think Describe works better since List calls are paginated.

Reply:
Even when using `describe`, excepting `ClientError` to mean it does not exist, there can be some other cause of ClientError right, since it's top level API error

Reply:
Fixed in latest revision